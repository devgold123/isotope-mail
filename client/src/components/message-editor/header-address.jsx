import React, {Component} from 'react';
import {translate} from 'react-i18next';
import PropTypes from 'prop-types';
import mainCss from '../../styles/main.scss';

export class HeaderAddress extends Component {
  constructor(props) {
    super(props);
    this.inputRef = React.createRef();
  }

  render() {
    const {id, className, chipClassName, label, addresses, onKeyPress, onBlur, onAddressRemove} = this.props;
    return (
      <div className={className} onClick={() => this.fieldClick()}>
        <label>{label}:</label>
        {addresses.map((a, index) => (
          <div key={index} className={`${chipClassName} ${mainCss['mdc-chip']}`} >
            <div className={mainCss['mdc-chip__text']}>{a}</div>
            <i onClick={() => onAddressRemove(id, index)} className={`material-icons ${mainCss['mdc-chip__icon']}
               ${mainCss['mdc-chip__icon--trailing']}`}>cancel</i>
          </div>
        ))}
        <input id={id} ref={this.inputRef} onKeyPress={onKeyPress} onBlur={onBlur} type={'email'} />
      </div>
    );
  }

  fieldClick() {
    this.inputRef.current.focus();
  }
}

HeaderAddress.propTypes = {
  t: PropTypes.func,
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  chipClassName: PropTypes.string,
  addresses: PropTypes.arrayOf(PropTypes.string),
  label: PropTypes.string,
  onKeyPress: PropTypes.func,
  onBlur: PropTypes.func,
  onAddressRemove: PropTypes.func
};

HeaderAddress.defaultProps = {
  className: '',
  chipClassName: '',
  addresses: [],
  label: '',
  onKeyPress: () => {},
  onBlur: () => {},
  onAddressRemove: () => {}
};

export default (translate()(HeaderAddress));
