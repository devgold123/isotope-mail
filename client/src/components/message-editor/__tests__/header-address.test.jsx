import React from 'react';
import {shallow} from 'enzyme/build/index';
import {HeaderAddress} from '../header-address';

describe('HeaderAddress component test suite', () => {
  test('Snapshot render, should render HeaderAddress', () => {
    const props = {
      t: translation => (translation),
      id: 'I\'mDown',
      className: 'Business',
      chipClassName: 'FirstClassChips',
      addresses: ['name@email.com', 'other@e-mail.mail'],
      label: 'BlackLabel12yrs',
      onKeyPress: () => {},
      onBlur: () => {},
      onAddressRemove: () => {}
    };
    const headerAddress = shallow(<HeaderAddress {...props} />);
    expect(headerAddress).toMatchSnapshot();
  });
});
