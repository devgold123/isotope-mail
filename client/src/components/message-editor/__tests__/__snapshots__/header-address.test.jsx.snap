// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderAddress component test suite Snapshot render, should render HeaderAddress 1`] = `
<div
  className="Business"
  onClick={[Function]}
>
  <label>
    BlackLabel12yrs
    :
  </label>
  <div
    className="FirstClassChips mdc-chip"
    key="0"
  >
    <div
      className="mdc-chip__text"
    >
      name@email.com
    </div>
    <i
      className="material-icons mdc-chip__icon
               mdc-chip__icon--trailing"
      onClick={[Function]}
    >
      cancel
    </i>
  </div>
  <div
    className="FirstClassChips mdc-chip"
    key="1"
  >
    <div
      className="mdc-chip__text"
    >
      other@e-mail.mail
    </div>
    <i
      className="material-icons mdc-chip__icon
               mdc-chip__icon--trailing"
      onClick={[Function]}
    >
      cancel
    </i>
  </div>
  <input
    id="I'mDown"
    onBlur={[Function]}
    onKeyPress={[Function]}
    type="email"
  />
</div>
`;
